CMakeLists.txt

```diff
@@ -108,6 +112,21 @@ find_package(UV REQUIRED)
 
 include(cmake/flags.cmake)
 
+if (WITH_TLS)
+    find_package(OpenSSL REQUIRED)
+
+    add_definitions(/DCPPHTTPLIB_OPENSSL_SUPPORT)
+
+    if (OPENSSL_FOUND)
+        include_directories(${OPENSSL_INCLUDE_DIR})
+        set(SOURCES_SSL_TLS src/3rdparty/clib-net/src/tls.c)
+    else()
+        message(FATAL_ERROR "OpenSSL NOT found: use `-DWITH_TLS=OFF` to build without TLS support")
```

Comment:
`find_package(OpenSSL REQUIRED)` should already fail, when OpenSSL cannot be found. The following if else is not needed. Just continue as if everything went well.

Reply:
Changed